### DISCUSSION TITLE
[Feature] Implement a tap-to-see-next/previous asset

### DISCUSSION THREAD

**Discussion by leleogere** [immich-app/immich#8155] [Feature Request]
When I have a lot of very similar pictures taken at the same moment, I usually want to keep only on or two of them. Unfortunately, seeing the differences between 2 photos by swiping between them is quite difficult because of the animation. Some photo viewers like F-Stop have a setting allowing the user to get to previous/next photo by simply tapping on the left/right side of the screen. It makes way easier to see the difference between 2 pictures to choose the best one. You can see in the video below the immich behaviour (you probably can't see the difference between the 2 pictures), and then the behaviour of F-Stop, in which the difference is much clearer.

https://github.com/immich-app/immich/assets/71326140/13149728-5ea4-4cf6-a5aa-2835d50d8bdf

Of course, this is a behaviour that's not completely intuitive, so I think it should be a setting disabled by default. 

Note that it does not prevent the default behaviour of hiding/showing the buttons tapping on the image, the user would just have to tap near the center to trigger this. The area to get to previous/next photo should only be rather thin on the sides.

**Comment by ThreeAurora** [comment]
Although they are busy, I do like this feature:)

**Comment by 1661410107114145152525** [comment]
I just want to highlight, that this is not a special expert feature for just the use case illustrated above where the web app is a valid workaround: **It is also really helpful for everyday use.**

Furthermore it just became really common in modern tools, to e.g. skim through Google Maps Images (or other popular apps), Stories on all major social platforms etc.

The main use case is to skip faster over images or navigate easier with on hand (especially if smartphone is held over your head), I would love to just tap on one of the sides. Just try the respective setting in F-Stop for reference as double-tap to zoom still works there while this is enabled.

Web already kinds of supports this via the arrows but they are really annoying when being always visible on small screens and the use case described in my text above is related to everyday use and not just a specific duplicate comparison.

This would be a real game changer usability wise. Thank you so much for reading this and considering it.

**Reply by 1661410107114145152525** [reply]
@leleogere Could you please add this use case to your description, so other users that are voting can quickly understand the advantages at once?

Furthermore: How can we raise visibility for this post again as it is already older than a year?

**Reply by leleogere** [reply]
Thanks for your support and sorry for answering so late! I will let the original post as is as a PR has already been opened for this feature.

**Comment by thezeroalpha** [comment]
Hey just FYI I'm working on this. I have a first version here that has the behavior, if you want to build your own: https://github.com/thezeroalpha/immich/tree/mobile/tap-to-next-previous-image
I need to figure out how to add a settings toggle for it and disable it by default, then I'll open a pull request. I'm not an immich dev, I just wanted this too and I decided to do it myself. So it'll take some time because I'm learning everything, but hopefully it'll make it into the app in the coming weeks.

**Reply by bo0tzz** [reply]
> I need to figure out how to add a settings toggle for it and disable it by default

I don't think that's necessary, right? Why shouldn't it always be on?

**Reply by 1661410107114145152525** [reply]
Same thoughts @bo0tzz. Settings toggle could be a separate feature request / follow-up if needed. :) @thezeroalpha

**Reply by thezeroalpha** [reply]
> Why shouldn't it always be on?

I was mostly following the logic of the original post: "this is a behaviour that's not completely intuitive, so I think it should be a setting disabled by default."

But OK! @bo0tzz you're a frequent contributor so I trust your opinion on it. I opened PR https://github.com/immich-app/immich/pull/20286, let's see what the feedback is there.

**Reply by 1661410107114145152525** [reply]
@thezeroalpha This is meanwhile the default for many popular apps like Google Maps, Instagram etc. so intuitivity shouldn't be an issue. :) Thanks for your effort!

**Reply by thezeroalpha** [reply]
So indeed there's some concerns about this being on by default -- if you want, comment your thoughts in https://github.com/immich-app/immich/pull/20286
